bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/datatypes/DataTypeMessage.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.comfoair.internal.datatypes;
+
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.comfoair.internal.ComfoAirCommandType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class to handle error messages
+ *
+ * @author Holger Hees - Initial Contribution
+ */
+public class DataTypeMessage implements ComfoAirDataType {
+
+    private Logger logger = LoggerFactory.getLogger(DataTypeMessage.class);
+
+    @Override
+    public State convertToState(int[] data, ComfoAirCommandType commandType) {
+
+        if (data == null || commandType == null) {
+            logger.trace("\"DataTypeMessage\" class \"convertToState\" method parameter: null");
+            return null;
+        } else {
+
+            int[] get_reply_data_pos = commandType.getGetReplyDataPos();
+
+            int errorAlo = data[get_reply_data_pos[0]];
+            int errorE = data[get_reply_data_pos[1]];
+            int errorEA = data[get_reply_data_pos[2]];
+            int errorAhi = data[get_reply_data_pos[3]];
+
+            String errorCode = "";
+
+            if (errorAlo > 0) {
+                errorCode = "A:" + convertToCode(errorAlo);
+            }
+
+            else if (errorAhi > 0) {
+                if (errorAhi == 0x80) {
+                    errorCode = "A0";
+                }
+                errorCode = "A:" + (convertToCode(errorAhi) + 8);
```

Comment:
What is the point of this if it gets overwritten anyway?

Reply:
This is unchanged from the OH1 binding. The code is from @HolgerHees, but following the protocol documentation there just seems to be an `else` missing. --> fixed